x/tokenfactory/keeper/admins.go

```diff
@@ -50,3 +50,22 @@ func (k Keeper) setAdmin(ctx context.Context, denom string, admin string) error
 
 	return k.setAuthorityMetadata(ctx, denom, metadata)
 }
+
+// GetDenomsFromAdmin returns all denoms for which the provided address is the admin
+func (k Keeper) GetDenomsFromAdmin(ctx context.Context, admin string) ([]string, error) {
```

Comment:
Add guard clause verification that the admin string is a valid AccAddress (I am good with admin being a string in the function sig since we are direct comparing later)

Then add test to cover as well for searching from a bad admin (should err not return [])

Reply:
As discussed on Discord, we will leave this as-is because we want to be able to search for denoms without an owner (e.g., '').